components/admin_console/system_users/system_users_dropdown/system_users_dropdown.jsx

```diff
@@ -134,7 +139,18 @@ export default class SystemUsersDropdown extends React.PureComponent {
 
     handleShowDeactivateMemberModal = (e) => {
         e.preventDefault();
-        this.setState({showDeactivateMemberModal: true});
+        if (this.shouldDisableBotsWhenOwnerIsDeactivated()) {
+            this.props.actions.loadBots(
+                Constants.Integrations.START_PAGE_NUM,
+                Constants.Integrations.PAGE_SIZE
+            ).then(
+                (result) => {
+                    if (result.data) {
+                        this.setState({loading: false, showDeactivateMemberModal: true});
```

Comment:
I think the state variable `loading` is not needed.

minor nit: Most of us are inclined to using `await` nowadays and we could use destructing with it as well making this an obvious pattern you would see around in the codebase.

```
const {data} = await func();
if(data) {
  setState({});
}
```